parts/windows/kuberneteswindowsfunctions.ps1

```diff
@@ -45,7 +45,7 @@ function DownloadFileOverHttp {
         $downloadTimer = [System.Diagnostics.Stopwatch]::StartNew()
         curl.exe -f --retry 5 --retry-delay 0 -L $Url -o $DestinationPath
         if (-not $?) {
-            throw "Curl exited with '$LASTEXITCODE' while attemping to downlaod '$Url'"
+            Set-ExitCode -ExitCode $global:WINDOWS_CSE_ERROR_DOWNLOAD_FILE -ErrorMessage "Curl exited with '$LASTEXITCODE' while attemping to downlaod '$Url'"

```

Comment:
Seems like we may use WINDOWS_CSE_ERROR_DOWNLOAD_FILE_WITH_RETRY instead?

Reply:
I actually changed this logic in aks-engine before the PR merged (but after if was copied here).
Powershell will think a .exe failed if it writes to stderr. 
curl.exe writes to stderr by even on success.
The correct behavior is to just check LASTEXITCODE

Reply:
https://github.com/Azure/aks-engine/pull/4327/files is what actually got checked into aks-engine

Reply:
Will change it with above change.